### ISSUE TITLE
Normalize zone name before test start

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-engine#739]
DNS does not make a distinction between lower case and upper case of ASCII letter, but Zonemaster-Engine preserves case through the machinery.

Change the API and implementation of Zonemaster-Engine so that it always down case the submitted zone name before doing any process.

If the zone name has any IDN A-labels those should also be in lower case.

A discussion is needed for the handling of any IDN U-labels in the zone name.

This is related to issue #734.

**Comment by mattias-p** [comment]
I completely agree that this should be done. I'm not sure it requires any API changes though. To me it feels more like a refactoring. In case we've forgotten another `lc` in some test case some bug might get fixed, but module/function signatures and intended behaviors should remain unaffected.

**Comment by matsduf** [comment]
What I meant by API was that if Engine returns the tested zone name in a formal string, then that might be in a different case. Maybe it is not needed.
